src/storage/keyvalue/EncodingPathStorage.ts

```diff
@@ -54,15 +54,46 @@ export class EncodingPathStorage<T> implements KeyValueStorage<string, T> {
    * Converts a key into a path for internal storage.
    */
   private keyToPath(key: string): string {
-    const encodedKey = Buffer.from(key).toString('base64');
-    return joinUrl(this.basePath, encodedKey);
+    return joinUrl(this.basePath, this.toBase64url(key));
   }
 
   /**
    * Converts an internal storage path string into the original path key.
    */
   private pathToKey(path: string): string {
-    const buffer = Buffer.from(path.slice(this.basePath.length), 'base64');
-    return buffer.toString('utf-8');
+    return this.fromBase64url(path.slice(this.basePath.length));
+  }
+
+  /**
+   * Encodes a string as base64url (a variant of base64 that is safe to use in URLs).
+   * Note that node 16 has built-in functions to support base64url encoding, meaning this function
+   * can be removed when node 16 becomes the minimum supported version for the CSS.
+   */
+  private toBase64url(input: string): string {
```

Comment:
Was first going to suggest moving these as util functions to StringUtil, but I realized just now that we can't do this. This has the same issue as hashing all paths: it breaks data migration for existing servers. If a resource is stored as `ABC/DEF` it will now not be found since we will be looking for `ABC_DEF`. So this will have to be reverted until we have a migration solution.